File: 121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] stocks) {
        int n = stocks.length;
        int min = stocks[0];
        int maxProfit = 0;
        // boolean closed = open;
        for(int i = 0; i < n; i++) { min = Math.min(min, stocks[i]); maxProfit = Math.max(maxProfit, stocks[i] - min);}
        return maxProfit;
    }
}

File: 121. Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        min_ = float('inf')
        max_profit = 0

        for price in prices:    
            min_ = min(min_, price)

            curr_profit = price - min_

            if curr_profit > max_profit:
                max_profit = curr_profit
        
        return max_profit

        
